lisa/sut_orchestrator/hyperv/platform_.py

```diff
@@ -334,3 +369,227 @@ def _delete_node(node_ctx: NodeContext, wait_delete: bool) -> None:
+            )
+
+        powershell = self._server.tools[PowerShell]
+        for cmd in confing_commands:
+            powershell.run_cmdlet(
+                cmdlet=cmd,
+                force_run=True,
+            )
+
+    def get_primary_nic_id(self) -> List[str]:
+        powershell = self._server.tools[PowerShell]
+        ip: str = self._server.public_address
+
+        # Get the NIC name via IP.
+        # We will get vEthernet switch interface name, not actual NIC for baremetal
+        cmd = (
+            "(Get-NetAdapter | Get-NetIPAddress | Where-Object "
+            f"{{ $_.IPAddress -eq '{ip}' }}).InterfaceAlias"
+        )
+        interface_name = powershell.run_cmdlet(
+            cmdlet=cmd,
+            force_run=True,
+        )
+
+        # Get the MAC for above interface
+        cmd = (
+            "(Get-NetAdapter | Where-Object "
+            f"{{ $_.Name -eq '{interface_name}' }}).MacAddress"
+        )
+        mac_address = powershell.run_cmdlet(
+            cmdlet=cmd,
+            force_run=True,
+        )
+
+        # Get all interfaces for above MAC Address
+        cmd = (
+            "(Get-NetAdapter | Where-Object "
+            f"{{ $_.MacAddress -eq '{mac_address}' }}).Name"
+        )
+        inf_names_str = powershell.run_cmdlet(
+            cmdlet=cmd,
+            force_run=True,
+        )
+        inf_names: List[str] = inf_names_str.strip().splitlines()
+
+        # Get device id for all above interface names we got
+        pnp_device_id_list: List[str] = []
+        for name in inf_names:
+            cmd = (
+                "(Get-NetAdapter | Where-Object "
+                f"{{ $_.Name -eq '{name}' }}).PnPDeviceID"
+            )
+            interface_device_id = powershell.run_cmdlet(
+                cmdlet=cmd,
+                force_run=True,
+            )
+            interface_device_id = interface_device_id.strip()
+            pnp_device_id_list.append(interface_device_id)
+
+        return pnp_device_id_list
+
+    def _set_device_passthrough_node_context(
+        self,
+        node_context: NodeContext,
+        node_runbook: HypervNodeSchema,
+        hv: HyperV,
+        vm_name: str,
+    ) -> None:
+        if node_runbook.device_passthrough:
+            hv.enable_device_passthrough(name=vm_name)
+
+            for config in node_runbook.device_passthrough:
+                devices = self.request_devices(
+                    pool_type=config.pool_type,
+                    count=config.count,
+                )
+                self._assign_devices_to_vm(
+                    vm_name=vm_name,
+                    devices=devices,
+                )
+                device_context = DevicePassthroughContext()
+                device_context.pool_type = config.pool_type
+                device_context.device_list = devices
+                node_context.passthrough_devices.append(device_context)
+
+    def configure_device_passthrough_pool(
```

Comment:
move public methods before private methods.

Reply:
@squirrelsc Done with new file created for hyperv device pool